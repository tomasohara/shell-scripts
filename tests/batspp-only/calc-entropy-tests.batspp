# Setup
$ bind 'set enable-bracketed-paste off'
$ unalias -a
$ for f in $(typeset -f | egrep '^\w+'); do unset -f $f; done
$ BIN_DIR=$PWD/..

## Bracketed Paste is disabled to prevent characters after output
## Example: 
# $ echo 'Hii'
# | Hi?2004l

# Tests for calc_entropy.perl and calc_entropy.py) scripts
# Settings
$ calc_entropy=calc_entropy.perl
$ simple="-simple"
$ verbose="-verbose"
# -or-: calc_entropy="calc_entropy.py"; simple="--simple"; verbose="--verbose"
# Convenience alias to use <NL> for entirely blank lines (i.e., no spaces)
$ alias encode-blank-line='perl -pe "s/^$/<NL>/;"'

$ $calc_entropy $simple .25 .25 .25 .25
Entropy
2.000

# Show input
$ head calc_entropy.*input* | encode-blank-line
==> calc_entropy.input <==
# Frequency of 'between' in Penn Treebank II WSJ annotations
24	pp-clr
6	pp-dir
4	pp-ext
35	pp-loc
2	pp-nom
7	pp-prd
42	pp-tmp
<NL>
==> calc_entropy.input2 <==
2	abc
4	def
1	ghi
1	jkl
<NL>
==> calc_entropy.simple.input <==
.25 .25 .25 .25
<NL>
==> calc_entropy.simple.input2 <==
.47 .42 .06 .05

# Simple input format

$ $calc_entropy $simple - < ./calc_entropy.simple.input
Entropy
2.000

$ $calc_entropy $simple - < ./calc_entropy.simple.input2
Entropy
1.497

# Verbose output

$ $calc_entropy $verbose $simple - < ./calc_entropy.simple.input2
#	prob	-p lg(p)    max p
#--------------------------------
#	0.470	0.512
#	0.420	0.526
#	0.060	0.244
#	0.050	0.216
#--------------------------------
# word	classes	freq	entropy	max_prob
#	1.000	1.497	   0.470
Entropy
1.497

# Usual input format

$ $calc_entropy - < ./calc_entropy.input
2.230

$ $calc_entropy $verbose - < calc_entropy.input | encode-blank-line
#		class	freq	prob	-p lg(p)
#		pp-tmp	42	0.350	0.530
#		pp-loc	35	0.292	0.518
#		pp-clr	24	0.200	0.464
#		pp-prd	7	0.058	0.239
#		pp-dir	6	0.050	0.216
#		pp-ext	4	0.033	0.164
#		pp-nom	2	0.017	0.098
#		total	120	1.000	2.230
<NL>
# word	classes	freq	entropy	max_prob
-	7	120	2.230	0.350
