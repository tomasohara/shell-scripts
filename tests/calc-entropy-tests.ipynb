{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Tests for calc_entropy.perl (or calc_entropy.py)\n",
    "#\n",
    "# NOTE: *** This is a test of the scripts, not of tomohara-aliases.bash, etc. ***\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?2004h\u001b[?2004l\r"
     ]
    }
   ],
   "source": [
    "# Setup\n",
    "bind 'set enable-bracketed-paste off'\n",
    "unalias -a\n",
    "for f in $(typeset -f | egrep '^\\w+'); do unset -f $f; done\n",
    "BIN_DIR=$PWD/.."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Bracketed Paste is disabled to prevent characters after output\n",
    "## Example: \n",
    "# $ echo 'Hii'\n",
    "# | Hi?2004l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "# Tests for calc_entropy.perl and calc_entropy.py) scripts\n",
    "# Settings\n",
    "calc_entropy=calc_entropy.perl\n",
    "simple=\"-simple\"\n",
    "verbose=\"-verbose\"\n",
    "# -or-: calc_entropy=\"calc_entropy.py\"; simple=\"--simple\"; verbose=\"--verbose\"\n",
    "#\n",
    "# Convenience alias to use <NL> for entirely blank lines (i.e., no spaces)\n",
    "alias encode-blank-line='perl -pe \"s/^$/<NL>/;\"'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Entropy\n",
      "2.000\n"
     ]
    }
   ],
   "source": [
    "$calc_entropy $simple .25 .25 .25 .25"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> calc_entropy.input <==\n",
      "# Frequency of 'between' in Penn Treebank II WSJ annotations\n",
      "24\tpp-clr\n",
      "6\tpp-dir\n",
      "4\tpp-ext\n",
      "35\tpp-loc\n",
      "2\tpp-nom\n",
      "7\tpp-prd\n",
      "42\tpp-tmp\n",
      "<NL>\n",
      "==> calc_entropy.input2 <==\n",
      "2\tabc\n",
      "4\tdef\n",
      "1\tghi\n",
      "1\tjkl\n",
      "<NL>\n",
      "==> calc_entropy.simple.input <==\n",
      ".25 .25 .25 .25\n",
      "<NL>\n",
      "==> calc_entropy.simple.input2 <==\n",
      ".47 .42 .06 .05\n"
     ]
    }
   ],
   "source": [
    "# Show input\n",
    "head calc_entropy.*input* | encode-blank-line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Simple input format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Entropy\n",
      "2.000\n"
     ]
    }
   ],
   "source": [
    "$calc_entropy $simple - < ./calc_entropy.simple.input"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Entropy\n",
      "1.497\n"
     ]
    }
   ],
   "source": [
    "$calc_entropy $simple - < ./calc_entropy.simple.input2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Verbose output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#\tprob\t-p lg(p)    max p\n",
      "#--------------------------------\n",
      "#\t0.470\t0.512\n",
      "#\t0.420\t0.526\n",
      "#\t0.060\t0.244\n",
      "#\t0.050\t0.216\n",
      "#--------------------------------\n",
      "# word\tclasses\tfreq\tentropy\tmax_prob\n",
      "#\t1.000\t1.497\t   0.470\n",
      "Entropy\n",
      "1.497\n"
     ]
    }
   ],
   "source": [
    "$calc_entropy $verbose $simple - < ./calc_entropy.simple.input2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Usual input format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2.230\n"
     ]
    }
   ],
   "source": [
    "$calc_entropy - < ./calc_entropy.input"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#\t\tclass\tfreq\tprob\t-p lg(p)\n",
      "#\t\tpp-tmp\t42\t0.350\t0.530\n",
      "#\t\tpp-loc\t35\t0.292\t0.518\n",
      "#\t\tpp-clr\t24\t0.200\t0.464\n",
      "#\t\tpp-prd\t7\t0.058\t0.239\n",
      "#\t\tpp-dir\t6\t0.050\t0.216\n",
      "#\t\tpp-ext\t4\t0.033\t0.164\n",
      "#\t\tpp-nom\t2\t0.017\t0.098\n",
      "#\t\ttotal\t120\t1.000\t2.230\n",
      "<NL>\n",
      "# word\tclasses\tfreq\tentropy\tmax_prob\n",
      "-\t7\t120\t2.230\t0.350\n"
     ]
    }
   ],
   "source": [
    "$calc_entropy $verbose - < calc_entropy.input | encode-blank-line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
