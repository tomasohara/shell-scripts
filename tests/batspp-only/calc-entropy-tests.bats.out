1..9
not ok 1 test-1
# (in test file batspp-only/calc-entropy-tests.bats, line 21)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== bind 'set enable-bracketed-paste off'\n ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-only/calc-entropy-tests.bats: line 28: bind: warning: line editing not enabled
# ========= $ unalias -a\n$ for f in ; do unset -f ; done\n$ BIN_DIR=/tmp/batspp-95453/test-1/.. =========
# $ unalias -a
# $ for f in bats_abort_timeout_countdown ()
# bats_add_debug_exclude_path ()
# bats_all_in ()
# bats_any_in ()
# bats_append_arrays_as_args ()
# bats_approx_realpath ()
# bats_binary_search ()
# bats_capture_stack_trace ()
# bats_check_status_from_trap ()
# bats_check_valid_version ()
# bats_create_test_tmpdirs ()
# bats_debug_trap ()
# bats_emit_trace ()
# bats_error_trap ()
# bats_evaluate_preprocessed_source ()
# bats_execute ()
# bats_exit_trap ()
# bats_extract_line ()
# bats_format_file_line_reference ()
# bats_format_file_line_reference_colon ()
# bats_format_file_line_reference_comma_line ()
# bats_format_file_line_reference_uri ()
# bats_frame_filename ()
# bats_frame_function ()
# bats_frame_lineno ()
# bats_generate_warning ()
# bats_get_child_processes_of ()
# bats_get_failure_stack_trace ()
# bats_internal_load ()
# bats_interrupt_trap ()
# bats_interrupt_trap_in_run ()
# bats_kill_childprocesses_of ()
# bats_load_library ()
# bats_load_library_safe ()
# bats_load_safe ()
# bats_merge_stdout_and_stderr ()
# bats_normalize_windows_dir_path ()
# bats_perform_test ()
# bats_prefix_lines_for_tap_output ()
# bats_preprocess_source ()
# bats_print_failed_command ()
# bats_print_stack_trace ()
# bats_quote_code ()
# bats_readlinkf ()
# declare -fx bats_readlinkf
# bats_redirect_stderr_into_file ()
# bats_replace_filename ()
# bats_require_minimum_version ()
# bats_separate_lines ()
# bats_setup_tracing ()
# bats_should_retry_test ()
# bats_sort ()
# bats_start_timeout_countdown ()
# bats_strip_string ()
# bats_teardown_trap ()
# bats_test_begin ()
# bats_test_function ()
# bats_timeout_trap ()
# bats_trim ()
# bats_trim_filename ()
# bats_version_lt ()
# bats_warn_minimum_guaranteed_version ()
# find_in_bats_lib_path ()
# get_mills_since_epoch ()
# load ()
# run ()
# setup ()
# skip ()
# teardown ()
# test-1-actual ()
# test-1-expected ()
# END_EXPECTED
# test-1-setup ()
# test-10-actual ()
# test-10-expected ()
# END_EXPECTED
# test-3-actual ()
# test-3-expected ()
# END_EXPECTED
# test-4-actual ()
# test-4-expected ()
# Entropy
# 2.000
# END_EXPECTED
# test-5-actual ()
# test-5-expected ()
# 24	pp-clr
# 6	pp-dir
# 4	pp-ext
# 35	pp-loc
# 2	pp-nom
# 7	pp-prd
# 42	pp-tmp
# 2	abc
# 4	def
# 1	ghi
# 1	jkl
# END_EXPECTED
# test-6-actual ()
# test-6-expected ()
# Entropy
# 2.000
# END_EXPECTED
# test-7-actual ()
# test-7-expected ()
# Entropy
# 1.497
# END_EXPECTED
# test-8-actual ()
# test-8-expected ()
# Entropy
# 1.497
# END_EXPECTED
# test-9-actual ()
# test-9-expected ()
# 2.230
# END_EXPECTED
# test_test-2d1 ()
# test_test-2d10 ()
# test_test-2d3 ()
# test_test-2d4 ()
# 2.000' "=========";
# test_test-2d5 ()
# test_test-2d6 ()
# 2.000' "=========";
# test_test-2d7 ()
# 1.497' "=========";
# test_test-2d8 ()
# Entropy
# 1.497' "=========";
# test_test-2d9 () ; do unset -f ; done
# $ BIN_DIR=/tmp/batspp-95453/test-1/..
# ============================
# /home/aveey/tom-project/shell-scripts/tests/batspp-only/calc-entropy-tests.bats: line 28: bind: warning: line editing not enabled
not ok 2 test-3
# (in test file batspp-only/calc-entropy-tests.bats, line 69)
#   `[ "$(test-3-actual)" == "$(test-3-expected)" ]' failed
# ========== # $ echo 'Hii'\n ==========
# ========= # | Hi?2004l =========
# # | Hi?2004l
# ============================
ok 3 test-4
ok 4 test-5
ok 5 test-6
ok 6 test-7
ok 7 test-8
ok 8 test-9
ok 9 test-10
